alfresco-search/doc/architecture/event-trackers/event-based-content-tracker-spike.md

```diff
@@ -0,0 +1,298 @@
+# Event Based Content Tracker for Alfresco
+
+![Completeness Badge](https://img.shields.io/badge/Document_Level-In_Progress-yellow.svg?style=flat-square)
+
+**Contents**
+
+- [Context](#context)
+- [How Text Extraction Works](#how-text-extraction-works)
+- [How the New Transform Service Works](#how-the-new-transform-service-works)
+- [Alternatives evaluation](#alternatives-evaluation)
+  - [1 - No Change to Search Service](#1---no-change-to-search-service)
+  - [2 - New Microservice in the Middle](#2---new-microservice-in-the-middle)
+  - [3 - Event Oriented Content Tracking](#3---event-oriented-content-tracking)
+- [Findings](#findings)
+- [References](#references)
+
+## Context
+
+The current approach of the Content Tracker in Search Service is to query Solr for any *dirty* documents which it then fetches from Alfresco. Once the content is successfully obtained from Alfresco, it marks it clean which eventually get committed to the index. This approach will need to be modified as it applies pressure on Alfresco on every call to get the content. Taking an event based approach where the Content Tracker subscribes to a *topic* with policy the specific behaviour will allow to get the extracted content when ready.
+
+There are currently different alternatives to be considered:
+
+1. No change to Search Service. The repository will off load the transformation to Transformation Service and pause the thread requesting the transformation until the content is transformed.
```

Comment:
its a call to repo to check if content exists if not it marks dirty and polls repo until it has the content which it then indexes it and marks it clean

Reply:
I think that currently is a sync invocation to Repository to get the Text Content of a Node:

https://github.com/Alfresco/SearchServices/blob/master/alfresco-search/src/main/java/org/alfresco/solr/SolrInformationServer.java#L2470

So in order to maintain Search Services code untouched, we need the repo to pause this thread until the transformation has been done. 

Correct me if I'm wrong.